### ISSUE TITLE
Cleanup: isSaleable() vs. isSalable() inconsistency

### ISSUE THREAD

**Issue by m-overlund** [OpenMage/magento-lts#1186]
### Summary (*)
There's seems to be a lot of inconsistencies around ``isSaleable`` functions.
It seem's like function ``isSalable()`` was created first, then later an alias ``isSaleable()`` was created to fix the spelling error.
But there's still defined functions with both names and calls to both functions in base code and templates.

In addition there is an alias ``getIsSalable()`` but ``getIsSaleable()`` does not exist.

### Examples (*)
https://github.com/OpenMage/magento-lts/search?q=isSaleable&unscoped_q=isSaleable
https://github.com/OpenMage/magento-lts/search?q=isSalable&unscoped_q=isSalable
https://github.com/OpenMage/magento-lts/search?q=getissalable&unscoped_q=getissalable

### Proposed solution

Name functions consistently and create aliases for the spelling error where due.

**Comment by seansan** [comment]
Is one of the 2 not already an alias of the other?

**Comment by sreichel** [comment]
I doubt this can be fixed completly w/o breaking stuff ....

Look at ...

        Mage::dispatchEvent('catalog_product_is_salable_after', array(
            'product'   => $this,
            'salable'   => $object
        ));

Fixing method names/adding aliases just fixes a part of the typo :(

**Comment by ADDISON74** [comment]
It is an issue that creates confusion for the developer but at the level of functionality it does not produce errors. Such a PR is more than a change of letters and can cause problems with extensions already used and no longer maintained by developers. Indeed the solution would be to correct the functions and create aliases for backward compatibility but what are the benefits for the future? Is anyone interested in doing a PR?

**Comment by fballiano** [comment]
we could add aliases but it's really a cosmetic change, at the end of the day everybody uses an IDE with autocompletion so...

**Comment by ADDISON74** [comment]
I agree, only a programmer who uses a text editor such as Notepad could still fill in the methods from memory. 

I suggest to convert this issue to discussion. It will remain for further opinions in the future.
